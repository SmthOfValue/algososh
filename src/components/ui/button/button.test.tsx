
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';

describe('Рендер Button', () => {
  it('Кнопка с текстом рендерится без ошибок', () => {
    const buttonWithText = renderer
      .create(<Button text="Проверочный текст" />)
      .toJSON();
      expect(buttonWithText).toMatchSnapshot();
  }); 

  it('Кнопка без текста рендерится без ошибок', () => {
    const buttonWithoutText = renderer
      .create(<Button />)
      .toJSON();
      expect(buttonWithoutText).toMatchSnapshot();
  }); 

  it('Кнопка в состоянии disabled рендерится без ошибок', () => {
    const buttonDisabled = renderer
      .create(<Button disabled />)
      .toJSON();
      expect(buttonDisabled).toMatchSnapshot();
  });

  it('Кнопка в состоянии loading рендерится без ошибок', () => {
    const buttonLoading = renderer
      .create(<Button isLoader={true} />)
      .toJSON();
      expect(buttonLoading).toMatchSnapshot();
  }); 
});

describe('Обработчик события нажатия кнопки', () => {
  it('Нажатие на кнопку вызывает корректный колбэк', () => {
    const callback = jest.fn();
    

    render(<Button text = 'Текст кнопки' onClick = {(callback)}/>)

    const button = screen.getByText("Текст кнопки");  

    fireEvent.click(button);        
    expect(callback).toHaveBeenCalled();
  }); 
});

