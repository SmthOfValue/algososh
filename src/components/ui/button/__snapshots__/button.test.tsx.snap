// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер Button Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Рендер Button Кнопка в состоянии disabled рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Рендер Button Кнопка в состоянии loading рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Рендер Button Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Проверочный текст
  </p>
</button>
`;
